clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ClusterCachingAgent.groovy

```diff
@@ -351,6 +355,18 @@ class ClusterCachingAgent implements CachingAgent, OnDemandAgent, AccountAware,
       .groupBy { it.autoScalingGroupName }
   }
 
+  private Map<String, List<Map>> loadLifecycleHooks(AmazonClients clients, List<String> autoScalingGroupNames) {
+    autoScalingGroupNames.collect { loadLifecycleHooks(clients, it) }*.groupBy { it.autoScalingGroupName } as Map<String, List<Map>>
```

Comment:
I'm a bit concerned about adding this in our caching run as it is going to be one API call per ASG (the rest of the describe calls we make are one per collection). I suspect this will trip us into throttling territory. Unfortunately not sure there is an alternative way to look these up in bulk.